test/nursery/autogluon_for_single_ts.py

```diff
@@ -0,0 +1,120 @@
+                zip(year_idx, month_of_year, day_of_week, hour_of_day, series.index.isin(holidays), target),
+                columns=['year_idx', 'month_of_year', 'day_of_week', 'hour_of_day', 'holiday', 'target'])
+            convert_type = {x: 'category' for x in df.columns.values[:4]}
+            df = df.astype(convert_type)
+            return df
+
+        for idx in range(len(list(dataset))):
+            ts = to_pandas(list(dataset)[idx])
+            df = get_prediction_dataframe(ts, lag)
+            output = self.ag_model.predict(df)  # type: dataframe
+            yield self.to_forecast(dataset, output, idx)
+
+    def to_forecast(self, gluonts_test, y_pred, i):
+        test = [list(gluonts_test)[i]]
+        forecasts = []
+        t = test[i]
+        start = to_pandas(t).index[-self.prediction_length]
+        y_hat = np.array(y_pred[i * self.prediction_length:(i + 1) * self.prediction_length])
+        samples = np.vstack([y_hat])  # * num_samples
+        sample = SampleForecast(freq=self.freq, start_date=pd.Timestamp(start, freq=self.freq), item_id=str(i),
+                                samples=samples)
+        forecasts.append(sample)
+        return forecasts
+
+
+class TabularEstimator(Estimator):
+    def __init__(self, freq, prediction_length):
+        super().__init__()
+        self.task = task
+        self.freq = freq
+        self.prediction_length = prediction_length
+
+    def train(self, training_data: Dataset) -> TabularPredictor:
+        def get_prediction_dataframe(series, lag=0, model=None, as_pandas=False):
+            hour_of_day = series.index.hour
+            month_of_year = series.index.month
+            day_of_week = series.index.dayofweek
+            year_idx = series.index.year
+            target = series.values
+            cal = calendar()
+            holidays = cal.holidays(start=series.index.min(), end=series.index.max())
+            df = pd.DataFrame(
+                zip(year_idx, month_of_year, day_of_week, hour_of_day, series.index.isin(holidays), target),
+                columns=['year_idx', 'month_of_year', 'day_of_week', 'hour_of_day', 'holiday', 'target'])
+            convert_type = {x: 'category' for x in df.columns.values[:4]}
+            df = df.astype(convert_type)
+            return df
+
+        df = get_prediction_dataframe(to_pandas(list(training_data)[0])[:-self.prediction_length])
+        ag_model = self.task.fit(df, label="target")
+        return TabularPredictor(ag_model, self.freq, self.prediction_length)
+
+
+def LocalTabularPredictor(*args, **kwargs):
+    return Localizer(TabularEstimator(*args, **kwargs))
+
+
+def stack_ts(dataset):
+    series = []
+    for i, entry in enumerate(list(dataset.test)):
+        series.append(to_pandas(entry))
+    return series
+
+
+if __name__ == '__main__':
+    from gluonts.dataset.repository.datasets import get_dataset
+
+    dataset = get_dataset("m4_weekly", regenerate=False)
+    series_ls = stack_ts(dataset)
```

Comment:
it doesn't look like this is used, so also `stack_ts` is maybe not needed?

Reply:
Yes. I will delete it.